packages/calling-stateful-client/src/CallClientState.ts

```diff
@@ -430,4 +430,82 @@ export interface CallClientState {
    * Completely controlled by the developer.
    */
   userId: CommunicationUserKind;
+  /**
+   * Stores the latest error for each API method.
+   *
+   * See documentation of {@Link CallErrors} for details.
+   */
+  latestErrors: CallErrors;
 }
+
+/**
+ * Errors teed from API calls to the Calling SDK.
+ *
+ * Each property in the object stores the latest error for a particular SDK API method.
+ *
+ * Errors from this object can be cleared using the TODO(implement me) {@Link newClearErrorsModifier}.
+ * Additionally, errors are automatically cleared when:
+ * - The state is cleared.
+ * - Subsequent calls to related API methods succeed.
+ * See documentation of individual stateful client methods for details on when errors may be automatically cleared.
+ */
+export type CallErrors = {
```

Comment:
nit: name might be confusing with CallErrors being an array of CallError

Reply:
This is an object that contains many errors, so the name seems fine.

We shouldn't encode type information in names (so no CallErrorMap or such). Will stick to this one.